core/src/test/java/org/frankframework/pipes/SoapWrapperPipeTest.java

```diff
@@ -516,13 +504,80 @@ void shouldUseConfiguredPipelineSoapVersionAuto() throws Exception {
 		outputWrapper.stop();
 	}
 
-	private void setupUnwrapAutoPipe() throws ConfigurationException, PipeStartException {
-		// Arrange - Simulate soapInputWrapper unwrapping the input and creating soapNamespace sessionKey using the namespace from the input
+	@Test
+	void shouldUnwrapEvenThoughSessionHasDifferentSoapVersionStored() throws Exception {
+		// Scenario:
+		// <inputWrapper direction="UNWRAP" storeResultInSessionKey="originalMessage"/> Note: input=1.2, stored version in session
+		// <inputWrapper soapVersion="1.1"> Note: output=1.1
+		// <outputWrapper direction="UNWRAP" removeOutputNamespaces="true" soapVersion="1.1"/> Note: input=1.1, should be unwrapped fine with 1.1
+
+		// Arrange: inputWrapper 1 - unwrap
+		setupUnwrapSoapPipe(SoapVersion.AUTO, false);
 		pipe.setDirection(Direction.UNWRAP);
-		pipe.setSoapVersion(SoapVersion.AUTO);
-		pipe.setName(PipeLine.INPUT_WRAPPER_NAME);
+		pipe.setStoreResultInSessionKey("originalMessage");
 		pipe.configure();
 		pipe.start();
+
+		// Arrange: inputWrapper 2 - wrap
+		P outputWrapper = createPipe();
+		autowireByType(outputWrapper);
+		outputWrapper.registerForward(new PipeForward("success", PipeLine.OUTPUT_WRAPPER_NAME + "2"));
+		outputWrapper.setName(PipeLine.OUTPUT_WRAPPER_NAME + "1");
+		outputWrapper.setSoapVersion(SoapVersion.SOAP11);
+		pipeline.addPipe(outputWrapper);
+		outputWrapper.configure();
+		outputWrapper.start();
+
+		// Arrange: outputWrapper 2 - unwrap
+		P outputWrapper2 = createWrapperSoapPipe(SoapVersion.SOAP11, false);
+		outputWrapper2.setName(PipeLine.OUTPUT_WRAPPER_NAME + "2");
+		outputWrapper2.setDirection(Direction.UNWRAP);
+		outputWrapper2.setRemoveOutputNamespaces(true);
+		outputWrapper2.registerForward(new PipeForward("success", "READY"));
+		outputWrapper2.configure();
+		outputWrapper2.start();
+		pipeline.addPipe(outputWrapper2);
+
+		session = new PipeLineSession();
+
+		// Act: message SOAP12 through pipes
+		PipeRunResult pipeRunResult1 = doPipe(pipe, Message.asMessage(soapMessageSoap12), session);
+		PipeRunResult pipeRunResult2 = doPipe(outputWrapper, pipeRunResult1.getResult(), session);
+		PipeRunResult pipeRunResult3 = doPipe(outputWrapper2, pipeRunResult2.getResult(), session);
+
+		// Assert: but the output is without SOAP namespaces, as it is unwrapped.
+		String result = pipeRunResult3.getResult().asString();
+		assertNotNull(result);
+		assertFalse(result.contains(SoapVersion.SOAP11.namespace), "result should be unwrapped for sure");
+		assertFalse(result.contains(SoapVersion.SOAP12.namespace), "result should be unwrapped for sure");
+		assertTrue(result.contains("FindDocuments_Response"), "result should contain original message, without namespaces");
+		outputWrapper.stop();
+	}
+
+	private P createWrapperSoapPipe(final SoapVersion soapVersion, boolean started) throws ConfigurationException, PipeStartException {
```

Comment:
I only see started being set to true so is it really needed to add a specific param for this?
What happens if it's not started?

Reply:
It is set to false in my new test: `shouldUnwrapEvenThoughSessionHasDifferentSoapVersionStored`
It is postponed in my test, so I can configure more variables/settings and then run the configure&start inside my unit test. Changing settings after a configure is not a good idea (and mostly not taken into account).